linera-core/src/worker.rs

```diff
@@ -286,12 +297,22 @@ pub struct WorkerState<StorageClient> {
     delivery_notifiers: Arc<Mutex<DeliveryNotifiers>>,
     /// The set of spawned [`ChainWorkerActor`] tasks.
     chain_worker_tasks: Arc<Mutex<JoinSet<()>>>,
+    /// The cache of running [`ChainWorkerActor`]s.
+    chain_workers: Arc<Mutex<LruCache<ChainId, ChainActorEndpoint<StorageClient>>>>,
 }
 
+/// The sender endpoint for [`ChainWorkerRequest`]s.
+type ChainActorEndpoint<StorageClient> =
+    mpsc::UnboundedSender<ChainWorkerRequest<<StorageClient as Storage>::Context>>;
```

Comment:
Consider creating an issue to revisit unbounded senders in general: backpressure might be better than the risk of running out of memory.

Reply:
Good idea, I opened #2176 for that :+1: